gm4_soul_fragments/data/gm4_soul_fragments/functions/player/search_fragment.mcfunction

```diff
@@ -0,0 +1,12 @@
+#@s = player that has lost soul fragments (@a[scores={gm4_sf_fragments=1..}] at @s)
+#from main
+
+#searches for fragments
+tag @s add gm4_sf_target
+scoreboard players operation #current gm4_sf_id = @s gm4_sf_id
+execute as @e[type=minecraft:area_effect_cloud,tag=gm4_soul_fragment,distance=..16] if score @s gm4_sf_id = #current gm4_sf_id at @s run function gm4_soul_fragments:soul_fragment/seek_player
```

Comment:
In a traditional server setting it would probably be faster to run this line directly from main, without the radius check.

Right now, this will look through the entities for every player, which might even have some duplicates in it, therefore I'd approach this differently:

1. In main, add a call that runs as and at all `gm4_soul_fragment` AECs
2. As each soul fragment, check for nearby players with a matching ID
3. If such a player is found, call your fancy particle and collection functions (This would also make the distance check in `soul_fragment/seek_player` line `16` redudant). Note that you already have such a call!